src/Sources.jl

```diff
@@ -17,6 +17,27 @@ Concrete subtypes of `AbstractSource` are:
 """
 abstract type AbstractSource end
 
+function check_github_archive(url::String)
+    # See
+    # * https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/
+    # * https://github.com/bazel-contrib/SIG-rules-authors/issues/11
+    # * https://github.com/spack/spack/issues/35250
+    # Note: according to
+    # <https://github.com/bazel-contrib/SIG-rules-authors/issues/11#issuecomment-1029861300>
+    # the `/archive/refs/tag` should be more stable, but that didn't happen in the incident
+    # on 2023-01-30, so it's unclear whether we can trust them.
+    if contains(url, r"github.com/[^/]+/[^/]+/archive/(refs/tag/)?[^/]+\.(tar\.gz|zip)$")
```

Comment:
This won't match manually uploaded release tarballs, right?

Reply:
Correct, those are `/download/`, like the file now used as test in https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/293/files#diff-5a745ed885ee2954c186a5c488dfb2f3d6c85e220cc63f5dc427035597ff0d1cR28